test/catch2/sub_level_error/unit/test_sub_level_error_drop_uncommitted_dirty.cpp

```diff
@@ -0,0 +1,167 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+#include "wt_internal.h"
+#include "../../wrappers/connection_wrapper.h"
+#include "../utils_sub_level_error.h"
+#include "../../../utility/test_util.h"
+
+/*
+ * [sub_level_error_drop_conflict]: test_sub_level_error_drop_conflict.cpp
+ * Tests the drop workflows that lead to EBUSY errors, and ensure that the correct sub level error
+ * codes and messages are stored.
+ */
+
+#define URI "table:test_drop_conflict"
+#define FILE_URI "file:test_drop_conflict.wt"
+#define UNCOMMITTED_DATA_MSG "the table has uncommitted data and cannot be dropped yet"
+#define DIRTY_DATA_MSG "the table has dirty data and can not be dropped yet"
+
+/*
+ * This test case covers EBUSY errors resulting from drop before committing/checkpointing changes.
+ */
+TEST_CASE("Test WT_UNCOMMITTED_DATA and WT_DIRTY_DATA", "[sub_level_error_drop_uncommitted_dirty]")
+{
+    std::string config = "key_format=S,value_format=S";
+    WT_SESSION *session = NULL;
+    WT_ERROR_INFO *err_info = NULL;
+    bool final = false;
+    bool mark_dead = false;
+    bool check_visibility = false;
+
+    connection_wrapper conn_wrapper = connection_wrapper(".", "create");
+    utils::prepare_session_and_error(&conn_wrapper, &session, &err_info);
+    REQUIRE(session->create(session, URI, config.c_str()) == 0);
+
+    WT_SESSION_IMPL *session_impl = (WT_SESSION_IMPL *)session;
+    REQUIRE(__wt_session_get_dhandle(session_impl, FILE_URI, NULL, NULL, 0) == 0);
+
+    SECTION("Test WT_UNCOMMITTED_DATA is not thrown")
```

Comment:
I was thinking if we could bunch the `not thrown` ones together, and have each call to the function with a comment  describing what we are testing e.g. 
```
SECTION("Test uncommitted visibility cases")
{
       REQUIRE(__wt_conn_dhandle_close(session_impl, final, mark_dead, check_visibility) == 0);
    
       utils::check_error_info(err_info, 0, WT_NONE, WT_ERROR_INFO_SUCCESS);
    
        // Test when visibility check is true.
        check_visibility = true;
        REQUIRE(__wt_conn_dhandle_close(session_impl, final, mark_dead, check_visibility) == 0);

        utils::check_error_info(err_info, 0, WT_NONE, WT_ERROR_INFO_SUCCESS);
}
```